src/arrays.js

```diff
@@ -7,33 +89,57 @@ const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
+  elements.forEach((element, i) => {
+    cb(element, i);
+  });
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
+  return elements.map((element, i) => cb(elements[i]));
 };
 
 const reduce = (elements, cb, memo) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+
+  if (memo === undefined) {
+    memo = 0;
+  }
+  return elements.reduce(cb, memo);
 };
 
 const find = (elements, cb) => {
   // Look through each value in `elements` and pass each element to `cb`.
   // If `cb` returns `true` then return that element.
   // Return `undefined` if no elements pass the truth test.
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i]) === true) {
+      return elements[i];
+    }
+  }
+  return undefined;
 };
 
+
 const filter = (elements, cb) => {
   // Similar to `find` but you will return an array of all elements that passed the truth test
   // Return an empty array if no elements pass the truth test
+  const newArr = [];
+  elements.forEach((element, i) => {
+    if (cb(elements[i])) newArr.push(elements[i]);
+  });
+  return newArr;
 };
 
 const flatten = (elements) => {
   // Flattens a nested array (the nesting can be to any depth).
   // Example: flatten([1, [2], [3, [[4]]]]); => [1, 2, 3, 4];
+  const flat = [].concat(...elements);
```

Comment:
Oh never mind, you can disregard my comment above on your other `flatten` implementation.